src/document/DocumentAnnotator.ts

```diff
@@ -1,39 +1,45 @@
-import debounce from 'lodash/debounce';
 import BaseAnnotator, { Options } from '../common/BaseAnnotator';
 import BaseManager from '../common/BaseManager';
+import HighlightPromoter from '../highlight/HighlightPromoter';
 import { centerRegion, isRegion, RegionManager } from '../region';
 import { Event } from '../@types';
 import { getAnnotation } from '../store/annotations';
 import { getSelection } from './docUtil';
 import { HighlightManager } from '../highlight';
-import { Mode, setSelectionAction } from '../store';
+import { Mode } from '../store';
 import { scrollToLocation } from '../utils/scroll';
 import './DocumentAnnotator.scss';
 
 // Pdf.js textLayer enhancement requires waiting for 300ms (they hardcode this magic number)
 const TEXT_LAYER_ENHANCEMENT = 300;
-// Selection change event is quite noisy, debounce it to avoid unnessary selection changes in store
-const SELECTION_CHANGE_DEBOUNCE = 100;
 
 export default class DocumentAnnotator extends BaseAnnotator {
     annotatedEl?: HTMLElement;
 
-    managers: Map<number, Set<BaseManager>> = new Map();
+    highlightPromoter?: HighlightPromoter;
 
-    selectionChangeTimer?: number;
+    managers: Map<number, Set<BaseManager>> = new Map();
 
     constructor(options: Options) {
         super(options);
 
+        if (this.isFeatureEnabled('highlightText')) {
+            this.highlightPromoter = new HighlightPromoter({
+                getSelection,
```

Comment:
It seems unnecessary to pass in `getSelection` as an option to the `HighlightPromoter` since we're trying to encapsulate highlight logic within the promoter class. Could it just be imported in `HighlightPromoter` directly?

Reply:
Like the above comment, `getSelection` is also related to pdfjs, which is only used in document annotator. We should implement different `getSelection` function for the other annotators.

Reply:
I gotcha, that makes sense to keep `HighlightPromoter` viewer agnostic